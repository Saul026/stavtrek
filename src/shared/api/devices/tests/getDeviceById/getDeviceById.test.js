const getDeviceById = require('./getDeviceById');
const axios = require('axios');

jest.mock('axios');

describe('getDeviceById', () => {
    let response;
    beforeEach(() => {
        response = {
            data: [
                {
                    id: 2,
                    attributes: {},
                    groupId: 0,
                    calendarId: 0,
                    protocol: null,
                    name: 'Тест',
                    uniqueId: '550393',
                    status: 'offline',
                    lastUpdate: '2023-11-27T07:47:46.000+00:00',
                    positionId: 131758,
                    phone: null,
                    model: null,
                    contact: null,
                    category: null,
                    disabled: false,
                    expirationTime: null,
                },
            ],
        };
    });
    test('Корректное значение', async () => {
        axios.get.mockReturnValue(response);
        const data = await getDeviceById(2);
        expect(data).toEqual(response);
        expect(data).toMatchSnapshot()
    });
});
