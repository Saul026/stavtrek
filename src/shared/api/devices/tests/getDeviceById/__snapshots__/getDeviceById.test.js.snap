// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDeviceById Корректное значение 1`] = `
{
  "data": [
    {
      "attributes": {},
      "calendarId": 0,
      "category": null,
      "contact": null,
      "disabled": false,
      "expirationTime": null,
      "groupId": 0,
      "id": 2,
      "lastUpdate": "2023-11-27T07:47:46.000+00:00",
      "model": null,
      "name": "Тест",
      "phone": null,
      "positionId": 131758,
      "protocol": null,
      "status": "offline",
      "uniqueId": "550393",
    },
  ],
}
`;
